internal/querycoord/meta.go

```diff
@@ -632,37 +630,37 @@ func (m *MetaReplica) saveGlobalSealedSegInfos(saves col2SegmentInfos) (col2Seal
 		}
 		col2SegmentChangeInfos[collectionID] = segmentsChangeInfo
 	}
+	/*
+		queryChannelInfosMap := make(map[UniqueID]*querypb.QueryChannelInfo)
+		for collectionID, segmentChangeInfos := range col2SegmentChangeInfos {
+			// get msgStream to produce sealedSegmentChangeInfos to query channel
+			queryChannelInfo := m.getQueryChannelInfoByID(collectionID)
+			msgPosition, err := m.sendSealedSegmentChangeInfos(collectionID, queryChannelInfo.QueryChannel, segmentChangeInfos)
+			if err != nil {
+				return nil, err
+			}
+			queryChannelInfo.SeekPosition = msgPosition
 
-	queryChannelInfosMap := make(map[UniqueID]*querypb.QueryChannelInfo)
-	for collectionID, segmentChangeInfos := range col2SegmentChangeInfos {
-		// get msgStream to produce sealedSegmentChangeInfos to query channel
-		queryChannelInfo := m.getQueryChannelInfoByID(collectionID)
-		msgPosition, err := m.sendSealedSegmentChangeInfos(collectionID, queryChannelInfo.QueryChannel, segmentChangeInfos)
-		if err != nil {
-			return nil, err
-		}
-		queryChannelInfo.SeekPosition = msgPosition
-
-		// update segmentInfo, queryChannelInfo meta to cache and etcd
-		seg2Info := make(map[UniqueID]*querypb.SegmentInfo)
-		for _, segmentInfo := range queryChannelInfo.GlobalSealedSegments {
-			segmentID := segmentInfo.SegmentID
-			seg2Info[segmentID] = segmentInfo
-		}
-		if infos, ok := saves[collectionID]; ok {
-			for _, segmentInfo := range infos {
+			// update segmentInfo, queryChannelInfo meta to cache and etcd
+			seg2Info := make(map[UniqueID]*querypb.SegmentInfo)
+			for _, segmentInfo := range queryChannelInfo.GlobalSealedSegments {
 				segmentID := segmentInfo.SegmentID
 				seg2Info[segmentID] = segmentInfo
 			}
-		}
+			if infos, ok := saves[collectionID]; ok {
+				for _, segmentInfo := range infos {
+					segmentID := segmentInfo.SegmentID
+					seg2Info[segmentID] = segmentInfo
+				}
+			}
 
-		globalSealedSegmentInfos := make([]*querypb.SegmentInfo, 0)
-		for _, info := range seg2Info {
-			globalSealedSegmentInfos = append(globalSealedSegmentInfos, info)
-		}
-		queryChannelInfo.GlobalSealedSegments = globalSealedSegmentInfos
-		queryChannelInfosMap[collectionID] = queryChannelInfo
-	}
+			globalSealedSegmentInfos := make([]*querypb.SegmentInfo, 0)
+			for _, info := range seg2Info {
+				globalSealedSegmentInfos = append(globalSealedSegmentInfos, info)
+			}
+			queryChannelInfo.GlobalSealedSegments = globalSealedSegmentInfos
+			queryChannelInfosMap[collectionID] = queryChannelInfo
+		}*/
```

Comment:
these lines (633-663) should be remove?

Reply:
Keep it for a short period of time in case of any meta lost due to removing querychannel.